src/System.Private.CoreLib/shared/System/Globalization/CultureData.Windows.cs

```diff
@@ -284,11 +287,11 @@ private string GetLanguageDisplayName(string cultureName)
 #else
             // Usually the UI culture shouldn't be different than what we got from WinRT except
             // if DefaultThreadCurrentUICulture was set
-            CultureInfo ci;
+            CultureInfo? ci;
 
             if (CultureInfo.DefaultThreadCurrentUICulture != null &&
                 ((ci = GetUserDefaultCulture()) != null) &&
-                !CultureInfo.DefaultThreadCurrentUICulture.Name.Equals(ci.Name))
+                !CultureInfo.DefaultThreadCurrentUICulture!.Name.Equals(ci!.Name))
```

Comment:
Similarly, why is the `!` on `ci` needed?  We also just checked that in the clause immediately prior.

Reply:
Not sure on either of these.  It's possible the assignment of `ci` inside the nested clause here is tripping up analysis but the `DefaultThreadCurrentUICulture` guard should be fine. @cston 

---
In reply to: [270457588](https://github.com/dotnet/coreclr/pull/23526#discussion_r270457588) [](ancestors = 270457588)

Reply:
Neither of these checks should be necessary with the latest Dev16 preview.

---
In reply to: [270471276](https://github.com/dotnet/coreclr/pull/23526#discussion_r270471276) [](ancestors = 270471276,270457588)

Reply:
Maybe they'll go away with consuming the new compiler bits?

Reply:
no longer required after updating compiler